src/headless/plugins/mam/index.js

```diff
@@ -47,12 +47,14 @@ converse.plugins.add('converse-mam', {
         });
         api.listen.on('enteredNewRoom', muc => muc.features.get('mam_enabled') && fetchNewestMessages(muc));
 
-        api.listen.on('chatReconnected', chat => {
+        const __fetchNewestMessages = chat => {
             // XXX: For MUCs, we listen to enteredNewRoom instead
             if (chat.get('type') === _converse.PRIVATE_CHAT_TYPE) {
                 fetchNewestMessages(chat);
             }
-        });
+        };
+        api.listen.on('chatReconnected', __fetchNewestMessages);
+        api.listen.on('chatBoxInitialized', __fetchNewestMessages);
 
```

Comment:
The `afterMessagesFetched` event is already used for newly created chats and AFAIK it should fire when you close a chat and reopen it. So the `chatBoxInitialized` event here is not the right one to listen to and we already listen to the right event below.

I suspect the messages aren't fetched because of ` && !_converse.connection.restored`.

Perhaps you can check?

Are you using BOSH or websocket?

`_converse.connection.restored` is usually only relevant if you're using BOSH and it will be true if you reloaded the page after logging in.

So if this is the cause, then you should also not be able to reproduce the bug without first reloading the page after logging in. Can you please check?

Reply:
> I suspect the messages aren't fetched because of ` && !_converse.connection.restored`.
> 
> Perhaps you can check?

Yes, you are right.



> Are you using BOSH or websocket?

I am using websocket.



> `_converse.connection.restored` is usually only relevant if you're using BOSH and it will be true if you reloaded the page after logging in.
> 
> So if this is the cause, then you should also not be able to reproduce the bug without first reloading the page after logging in. Can you please check?

Yes, you are entirely right.

Reply:
I have modified my PR.
I propose to take into account 'clear_messages_on_reconnection' setting into 'afterMessagesFetched' event listener.